### PR TITLE
Prevent multi-threading race conditions and add a dunce feature

### PR DESCRIPTION
**Pull request by bteitler**
Prevent multi-threading race conditions with a global lock.  The convention is to add a lock around any user command or event handler (anything with a discord tag).  Utility functions should not a lock to prevent deadlock.  Consider using re-entrant locks in the future if this gets to complicated.

Add a 'dunce' feature to keep track of trouble makers and lightly shame them without all the bad feelings that come with not being able to actually play.  Currently replaces achievement list with a single dunce cap image and shows the dunce reason.  Also forces rating to be shown.

### CODE REVIEW

File: variables.json [hunk 1]
```diff
@@ -10,7 +11,7 @@
     "rank9": "<:rank9:1000673940173242378>",
     "rank10": "<:rank10:1000673949761417306>",
     "rankS": "<:rankS:1003216122205708359>",
-    "TOKEN": "NTY0MjU2NzQxNTIxMzU4ODU4.GYBmE2.5kCc33t6vXEA5jHJq8rJ2GQ_IbL8tvfDLMf4Bo",
+    "TOKEN": "NTY0MjU2NzQxNTIxMzU4ODU4.GNn1Ov.8zjNhtwXDVj9Grf2uyaacJdhQTsTf4vxOCxYvc",
```
Comment (ededdneddyfan) [inline variables.json:14]: i assume we replaced this token right?

**Review by ededdneddyfan — APPROVED** [review]
Looks good, concerned about committing a sensitive token to github but I remember seeing coach chatting about this

File: variables.json (hunk 1 above)
Comment (12souza) [reply variables.json:14]: ya discord immediately deactivates the bot and DMs me about the token.. so cheese uploaded this around 4:40am or so PST and i was able to get it back up at 5:30am.  However i was just lucky to be awake lol.. I would just replace the token  upon upload with "TOKEN"

Comment (bteitler) [reply variables.json]: Should be fixed now.

**Pull request merged**